fe/fe-core/src/main/java/com/starrocks/transaction/TxnFinishState.java

```diff
@@ -67,4 +74,46 @@ public void fromBytes(byte[] bytes) throws IOException {
             abnormalReplicasWithVersion.put(pb.abnormalReplicasWithVersion.get(i), pb.abnormalReplicasWithVersion.get(i + 1));
         }
     }
+
+    public String toJson() throws IOException {
```

Comment:
Why not use GSON? What's the advantage to write json deserialize by hand?

Reply:
done